tests/codeception/functional/MegaMenuCest.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+use Drupal\Core\Url;
+use Faker\Factory;
+
+/**
+ * Class MegaMenuCest.
+ *
+ * @group install
+ */
+class MegaMenuCest {
+
+  /**
+   * Faker service.
+   *
+   * @var \Faker\Generator
+   */
+  protected $faker;
+
+  /**
+   * Test constructor.
+   */
+  public function __construct() {
+    $this->faker = Factory::create();
+  }
+
+  public function _before(FunctionalTester $I) {
+    $I->resizeWindow(2000, 1400);
+    $I->logInWithRole('administrator');
+    $I->amOnPage('/admin/config/site-options');
+    $I->see('Enable New Mega Menu');
+
+    $this->megaMenuEnabled = (bool) $I->grabAttributeFrom('[name="field_en_mega_menu[value]"]', 'checked');
+    if (!$this->megaMenuEnabled) {
+      $I->checkOption('Enable New Mega Menu');
+      $I->click('Save');
+      drupal_flush_all_caches();
+    }
```

Comment:
It's best to not rely on the UI for `_before` or `_after` methods. Use the drupal API in case something in the browser fails.

Reply:
Moved this back into the main function. Long ago, I tried to use the Drupal API to set the config pages setting to enable the mega menu, but that didn't work.